spyder/utils/programs.py

```diff
@@ -710,7 +712,27 @@ def run_python_script_in_terminal(fname, wdir, args, interact,
                 else:
                     run_program(program['cmd'], arglist)
                 return
-        # TODO: Add a fallback to OSX
+    elif sys.platform == 'darwin':
+        f = tempfile.NamedTemporaryFile('wt', prefix='run_spyder_',
+                                        suffix='.sh', dir=get_temp_dir(),
+                                        delete=False)
+        f.write(' '.join(p_args))
+        f.close()
+        os.chmod(f.name, 0o777)
+
+        def run_terminal_thread():
+            if wdir:
+                proc = run_shell_command('open -a Terminal.app ' + f.name,
+                                         cwd=wdir)
+            else:
+                proc = run_shell_command('open -a Terminal.app ' + f.name)
+            # Prevent race condition
+            time.sleep(2)
+            proc.wait()
+            os.remove(f.name)

```

Comment:
Would it be better to then not remove these files?

They will be stored in a temp folder so the user can delete this folder any time it is needed (which for these small files will take a very very long time anyway)

I agree with @ccordoba12 that 2 seconds is just too much.

Reply:
The problem is that the folder for these temporal files is stored in a place not that easy to track for an user, and every time they run the program from the external console a file will be created. At the end, a lot of files will remain on the system. 

If you try it, that time will delay the deletion of the file rather than waiting 2 seconds to open the terminal.

Reply:
Is this blocking the UI? or the 2 second wait is on a thread?

Reply:
Is on the thread, the UI isn't blocked

Reply:
Ok then all good :-) !

Reply:
But I would like to have some tests added to this :-)

The "script" could be the creation of a file with some content so that we can check that happens after some time.

Reply:
I'm not sure of how to create the test, because there's not a way to know if the terminal is open or not.

Reply:
@steff456 I suggest we run a script that does some operation that ca be checked not directly check if a terminal is open or not, so:

```python
def run_python_script_in_terminal(fname, wdir, args, interact,
                                  debug, python_args, executable=None):
```

where:

```python
# fname
with open('/Users/gpena-castellanos/Desktop/somefile.txt', 'w') as f:
    f.write('This is a test')
```

And we check that the file is created and that the contents match.